app/src/main/java/com/ihunda/android/binauralbeat/BBeat.java

```diff
@@ -894,6 +929,16 @@ public void onClick(View v) {
                     public void onClick(View v) {
                         StartPreviouslySelectedProgram();
                         removeDialog(DIALOG_PROGRAM_PREVIEW);
+                        try {
```

Comment:
This code should be in handled in the goToState function as it's the master function to handle app state. There's a section for going from SETUP to IN PROGRAM state and there we should init History.

Reply:
okay.
Done